refactor: centralize network defaults in YAML with bash generation scripts

ludamad opened the pull request:
## Summary

- Centralize yarn-project default network configuration in a single YAML source of truth: `spartan/environments/network-defaults.yml`
- Generate TypeScript configs from YAML using bash scripts with `yq` and `jq` for:
  - CLI network configs (`yarn-project/cli/src/config/generated/networks.ts`)
  - L1 contracts defaults (`yarn-project/ethereum/src/generated/l1-contracts-defaults.ts`)
  - Slasher defaults (`yarn-project/slasher/src/generated/slasher-defaults.ts`)
- Use YAML anchors for DRY configuration inheritance between networks

ludamad commented on spartan/environments/default.json:
```diff
@@ -0,0 +1,117 @@
+{
+  "_comment": "Spartan Default Configuration - This is the canonical source of truth for network defaults. Network-specific configs can override any of these values.",
```
Just an MVP incarnation of a central location for the previously dispersed defaults

alexghr replied:
if we use yaml.. comments would be free

alexghr replied:
Oh but this way we can JSON.parse it directly inside Node scripts.

alexghr commented on spartan/environments/next-net.env:
```diff
@@ -4,7 +4,34 @@ CLUSTER=aztec-gke-private
+# ============================================
+# Core settings
+L1_CHAIN_ID=11155111
 ETHEREUM_CHAIN_ID=11155111
```
duplicate?

alexghr commented on yarn-project/cli/src/config/chain_l2_config.ts:
```diff
@@ -572,19 +701,63 @@ function getDefaultDataDir(networkName: NetworkNames): string {
-  enrichEnvironmentWithChainConfig(config);
+  // For real networks, L2ChainConfig values MUST come from environment variables
+  // (spartan .env files). No defaults are applied - this ensures config matches
+  // deployed contracts and prevents silent misconfigurations.
```
we lose the ability to `--network testnet`?

alexghr replied:
Oh the responsibility was moved.

alexghr reviewed (approved):
Looks great!

ludamad commented:
Re the json - lets live with the pseudo-comment for now. I want this to move to typescript as the ultimate convenient format :)

alexghr commented on yarn-project/cli/scripts/generate.sh:
```diff
@@ -0,0 +1,35 @@
+// Auto-generated from spartan/environments/network-defaults.yml
+// Do not edit manually - run yarn generate to regenerate
+
+export type NetworkConfigEnv = Record<string, string | number | boolean>;
```
Minor thing: I think we'd be better off in terms of typechecking with a marking the config `as const`?

```
export const ${network}Config = /* yq soup */ as const;
```

ludamad replied:
Yes good catch I saw this had a type and thought Claude did something reasonable :D

ludamad committed:
refactor: centralize network defaults in YAML with bash generation scripts

## Summary

- Centralize yarn-project default network configuration in a single YAML source of truth: `spartan/environments/network-defaults.yml`
- Generate TypeScript configs from YAML using bash scripts with `yq` and `jq` for:
  - CLI network configs (`yarn-project/cli/src/config/generated/networks.ts`)
  - L1 contracts defaults (`yarn-project/ethereum/src/generated/l1-contracts-defaults.ts`)
  - Slasher defaults (`yarn-project/slasher/src/generated/slasher-defaults.ts`)
- Use YAML anchors for DRY configuration inheritance between networks
